slave.mk

```diff
@@ -198,6 +198,14 @@ ifeq ($(BUILD_LOG_TIMESTAMP),)
 override BUILD_LOG_TIMESTAMP := $(DEFAULT_BUILD_LOG_TIMESTAMP)
 endif
 
+ifeq ($(NUM_ASIC),)
+override NUM_ASIC := $(DEFAULT_NUM_ASIC)
```

Comment:
>NUM_ASIC [](http://example.com/codeflow?start=9&length=8)

NUM_ASIC is a runtime parameter. The same sonic image could run either 6-ASIC switch or 66-ASIC switch (just example). Why the build process need to know the number?

Reply:
That is right, this is mainly done to make the building easier in Azure Pipeline script. If we go with the current build setup,
We will need to something like this:
```
          set -ex
          if [ $(GROUP_NAME) == vs ]; then
            sudo echo NUM_ASIC=$(num_asic) > ./device/virtual/x86_64-kvm_x86_64-r0/asic.conf
            asic_id=0
            while [ $asic_id -le $(num_asic) ]; do
               sudo echo DEV_ID_ASIC_$asic_id=$asic_id >> ./device/virtual/x86_64-kvm_x86_64-r0/asic.conf
               asic_id=$(( $asic_id + 1 ))
            done
            make $BUILD_OPTIONS target/sonic-vs.img.gz && mv target/sonic-vs.img.gz target/sonic-masic-vs.img.gz
            sudo echo NUM_ASIC=1 > ./device/virtual/x86_64-kvm_x86_64-r0/asic.conf
            rm ./target/debs/buster/sonic-device-data_1.0-1_all.deb
            make $BUILD_OPTIONS target/docker-sonic-vs.gz target/sonic-vs.img.gz target/docker-ptf.gz
```
If we have a build parameter, then we could directly build multi-asic VS image as in the PR:
https://github.com/Azure/sonic-buildimage/pull/7873/

Reply:
Now I understand you are using the parameter only for KVM image. Could you change parameter name to sth like `NUM_VIRTUAL_ASIC` or `NUM_ASIC_VS` to make it explicit?

Reply:
Updated as suggested.